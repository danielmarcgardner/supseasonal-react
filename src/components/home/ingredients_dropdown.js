import React , { Component } from 'react';
import { Dropdown, Button } from 'semantic-ui-react';
import { setIngredient, allIngredients } from '../../actions/index';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

const mapStateToProps = (state, ownProps) => {
  return {
    ingredients: state.ingredients,
    ingredient: state.ingredient
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ setIngredient, allIngredients }, dispatch)
}

const dropDownMaker = (ingredients) => {
  return ingredients.map((ing) => {
    let obj = {}
      obj.value = ing.id;
      obj.text = ing.food_name;
  })
}



const IngredientSearch = () => (
  <Button color='orange' content="Search By Ingredient"/>
)

class IngredientsDropdown extends Component {
  componentWillMount(){
    this.props.allIngredients()
  }
  render() {
    if (!this.props.ingredients) {
      return <div>LOADING</div>
    }
    return (
      <form onSubmit={(event) => {
        event.preventDefault()
        this.props.setIngredient(event.target.value)
      }}>
        <div>
          <Dropdown placeholder='Select An Ingredient' fluid selection options={dropDownMaker(this.props.ingredients)} />
        </div>
        <div>
          <IngredientSearch type="submit"><Link to={`/${this.props.ingredient}`}>Search By Ingredients</Link></IngredientSearch>
        </div>
      </form>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(IngredientsDropdown)
