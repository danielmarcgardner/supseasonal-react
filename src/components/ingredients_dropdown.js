import React , { Component } from 'react';
import { Dropdown } from 'semantic-ui-react'


const DropdownIngredientsSelection = () => {(
  <Dropdown placeholder='Select A Month' fluid selection options={friendOptions} />
)}


// class IngredientsDropdown extends Component {
//   // let navitems = food_name.map((foodnames) => <NavItem food_name={foodnames} />)
//   componentWillMount() {
//           $(document).ready(() => {
//               $('.collapsible').collapsible();
//           });
//       }
//
//   render() {
//     return (
//       <div>
//         <div>
//         <Dropdown  className="collapsible" trigger={
//         <Button>Select A Food!</Button>
//         }>
//           <NavItem>Food 1</NavItem>
//           <NavItem>Food 2</NavItem>
//           <NavItem>Food 3</NavItem>
//         </Dropdown>
//         </div>
//         <div>
//           <IngredientSearch />
//         </div>
//       </div>
//     )
//   }
// }
//
// export default IngredientsDropdown
