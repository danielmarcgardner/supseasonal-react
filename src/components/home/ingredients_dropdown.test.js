import {IngredientsDropdown} from './ingredients_dropdown';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('IngredientsDropdown', () => {
  const props = {
    allIngredients: function() {
      console.log('This function was called');
    },
    ingredient: 1,
    ingredients: [
      {id: 1, food_name: "artichokes"},
      {id: 2, food_name: "arugula"},
      {id: 3, food_name: "asparagus"},
    ]
  };

  // it('renders', () => {
  //   const div = document.createElement('div');
  //   ReactDOM.render(<IngredientsDropdown {...props}/>, div);
  // });

  test('snapshots', () => {
    const component = shallow(
      <IngredientsDropdown {...props} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
