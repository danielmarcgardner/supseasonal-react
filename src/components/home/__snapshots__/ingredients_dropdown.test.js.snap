exports[`IngredientsDropdown snapshots 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="dropdownIng">
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      fluid={true}
      icon="dropdown"
      noResultsMessage="No results found."
      onChange={[Function]}
      openOnFocus={true}
      options={
        Array [
          Object {
            "text": "Artichokes",
            "value": 1,
          },
          Object {
            "text": "Arugula",
            "value": 2,
          },
          Object {
            "text": "Asparagus",
            "value": 3,
          },
        ]
      }
      placeholder="Select An Ingredient"
      renderLabel={[Function]}
      selectOnBlur={true}
      selection={true} />
  </div>
  <div
    className="SearchButton">
    <Link
      replace={false}
      to="/ingredient/1">
      <Button
        as="button"
        color="orange">
        Search By Ingredients
      </Button>
    </Link>
  </div>
</form>
`;
